ldes-server-port-ingest/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/ingest/MemberIngesterImpl.java

```diff
@@ -26,30 +31,30 @@ public MemberIngesterImpl(MemberIngestValidator validator, MemberRepository memb
 		this.eventPublisher = eventPublisher;
 	}
 
-	@Override
-	public void ingest(Member member) {
-		validator.validate(member);
-		final boolean isNewMember = !memberRepository.memberExists(member.getId());
-		final String memberId = member.getId().replaceAll("[\n\r\t]", "_");
-		if (isNewMember) {
-			ingestNewMember(member, memberId);
-		} else {
-			log.warn("Duplicate member ingested. Member with id {} already exist", memberId);
-		}
-	}
-
-	private void ingestNewMember(Member member, String memberId) {
-		Metrics.counter("ldes_server_ingested_members_count").increment();
-		final Member savedMember = save(member);
-		final var memberIngestedEvent = new MemberIngestedEvent(savedMember.getModel(), savedMember.getId(),
-				savedMember.getCollectionName(), savedMember.getSequenceNr());
-		eventPublisher.publishEvent(memberIngestedEvent);
-		log.debug("Member with id {} ingested.", memberId);
-	}
-
-	private Member save(Member member) {
-		member.removeTreeMember();
-		return memberRepository.saveMember(member);
-	}
-
-}
+    @Override
+    public void ingest(Member member) {
+        validator.validate(member);
+        final String memberId = member.getId().replaceAll("[\n\r\t]", "_");
+        ingestNewMember(member, memberId);
+    }
+
+    private void ingestNewMember(Member member, String memberId) {
+        Optional<Member> savedMember = insert(member);
+        if (savedMember.isPresent()) {
+            Member sMember = savedMember.get();
+            Metrics.counter(LDES_SERVER_INGESTED_MEMBERS_COUNT).increment();
+            final var memberIngestedEvent = new MemberIngestedEvent(sMember.getModel(), sMember.getId(),
+                    sMember.getCollectionName(), sMember.getSequenceNr());
+            eventPublisher.publishEvent(memberIngestedEvent);
+            log.debug(MEMBER_WITH_ID_INGESTED, memberId);
+        } else {
+            log.warn(DUPLICATE_MEMBER_INGESTED_MEMBER_WITH_ID_ALREADY_EXISTS, memberId);
+        }
+    }
```

Comment:
Maybe I am nitpicking here a little, but I see the public method only has two lines of code, where only the memberId as extracted and passed to the private function, so I would rewrite it a little.
```java
    @Override
    public void ingest(Member member) {
        validator.validate(member);
        final String memberId = member.getId().replaceAll("[\n\r\t]", "_");
	insert(member).ifPresentOrElse(
			insertedMember-> handleSuccessfullMemberInsertion(member, memberId),
			() -> log.debug(DUPLICATE_MEMBER_INGESTED_MEMBER_WITH_ID_ALREADY_EXISTS, memberId)
	);
    }

    private void handleSuccessfullMemberInsertion(Member member, String memberId) {
    	Metrics.counter(LDES_SERVER_INGESTED_MEMBERS_COUNT).increment();
	final var memberIngestedEvent = new MemberIngestedEvent(member.getModel(), member.getId(),
			member.getCollectionName(), member.getSequenceNr());
	eventPublisher.publishEvent(memberIngestedEvent);
	log.debug(MEMBER_WITH_ID_INGESTED, memberId);
    }
```
This way you can also see in the public method that we try to insert a new member in the repo and we can see what happens if it succeeds or fails